Source/JavaScriptCore/dfg/DFGClobberize.h

```diff
@@ -1952,7 +1952,8 @@ void clobberize(Graph& graph, Node* node, const ReadFunctor& read, const WriteFu
         Edge& keyEdge = node->child2();
         AbstractHeapKind heap = (mapEdge.useKind() == WeakMapObjectUse) ? JSWeakMapFields : JSWeakSetFields;
         read(heap);
-        def(HeapLocation(WeakMapGetLoc, heap, mapEdge, keyEdge), LazyNode(node));
+        if (keyEdge.useKind() == ObjectUse || keyEdge.useKind() == SymbolUse)
+            def(HeapLocation(WeakMapGetLoc, heap, mapEdge, keyEdge), LazyNode(node));
```

Comment:
Why can't we always def here, even for CellUse? If we have a string or registered symbol, its result is false